lib/os/onoff.c

```diff
@@ -289,7 +301,7 @@ static void onoff_stop_notify(struct onoff_service *srv,
 	 * In either case remove the last reference, and notify all
 	 * waiting clients of operation completion.
 	 */
-	if (res < 0) {
+	if ((res < 0) && (res != -ECANCELED)) {
```

Comment:
`-ECANCELED` is not error from the service but from cancellation. In that case, it can transition to off state as if successful stop would occur. If new requests came in the meanwhile it will behave the same way as if they came during stopping.